samza-kv/src/main/scala/org/apache/samza/storage/kv/KeyValueStorageEngine.scala

```diff
@@ -200,6 +200,11 @@ class KeyValueStorageEngine[K, V](
 
     // flush the store and the changelog producer
     flush() // TODO HIGH pmaheshw SAMZA-2338: Need a way to flush changelog producers. This only flushes the stores.
+
+    if (Thread.currentThread().isInterrupted) {
+      warn("Received an interrupt during store restoration. Exiting without restoring the full state.")
+      throw new InterruptedException("Received an interrupt during store restoration.")
+    }
```

Comment:
Could anything in line 190-202 check the isInterrupted flag and then throw the InterruptedException? Would there be any issue if that happened?
It seems like this block should be moved up earlier to provide a quick exit, but I'm not sure if you intentionally wanted to run the above lines. For example, you still have a `doPutAll` that will not get interrupted explicitly.

Reply:
Currently, there is none. Even if we were to do it, the upstream is agnostic to where inside the restore logic `InterruptedException` is thrown. The interrupt is more of a best effort notification to the implementors to safely abort whatever they are doing. For this implementation,  I chose put this below to make sure we write the records we already picked up since `doPutAll` by default does async writes which are faster and also the writes only handles the last batch if it is not.

What do you think?

Reply:
I was wondering if you depended on the interrupt being checked in this block because you wanted to ensure that `flush` did get called.
An example situation I was thinking of was that RocksDB does do interrupt handling in `doPutAll` (maybe not today, but it's still possible in the future), and that could maybe cause `flush` to get skipped.
I just wanted to make sure all the interrupt situations are handled. If you can confirm that this will handle interrupt situations ok, then I am good with this.

Reply:
Synced up offline. Yes it is.